script.js

```diff
@@ -0,0 +1,746 @@
+            if (loopCount < app.state[region].laneData.length - 1) {
+                loopCount++;
+                var laneData = app.state[region].laneData[loopCount];
+                app.allLanesQueue(url, region);
+            } else {
+                var arrayR = app.display.regionsArray;
+                arrayR.shift();
+                app.display.regionsArray = arrayR;
+                app.allLanesQueue(url, arrayR);
+            }
+        }).fail(function(){
+            console.log("*** ajax fail T ***");
+        });
+    },
+
+    // ======= initMap =======
+    initMap: function () {
+        console.log("app.initMap");
+        L.mapbox.accessToken = "pk.eyJ1IjoiYWx1bHNoIiwiYSI6ImY0NDBjYTQ1NjU4OGJmMDFiMWQ1Y2RmYjRlMGI1ZjIzIn0.pngboKEPsfuC4j54XDT3VA";
+        var map = L.mapbox.map(
+            app.map.mapEl,
+            app.map.mapStyle,
+            { zoomControl: false })
+                .setView([app.map.centerLat, app.map.centerLng], app.map.zoom);
+        new L.Control.Zoom({ position: "bottomright" }).addTo(map);
+        return map;
+    },
+
+    // ======= activateMap =======
+    activateMap: function() {
+        console.log("activateMap");
+
+        // ======= mouse location =======
+        app.activeMap.on("mousemove", function (e) {
+            document.getElementById("lat").innerHTML = (e.latlng.lat).toFixed(4);
+            document.getElementById("lng").innerHTML = (e.latlng.lng).toFixed(4);
+        });
+
+        // ======= mouse location =======
+        app.activeMap.on("click", function (e) {
+            console.log("\n-- mapClick -- ");
+            var latLng = [JSON.stringify(e.latlng.lat), JSON.stringify(e.latlng.lng)];
+            !$("#startLoc").val() ?
+                app.addRouteMarker("start", latLng) :
+                app.addRouteMarker("end", latLng);
+        });
+
+    },
+
+    // ======= addRouteMarker =======
+    addRouteMarker: function(startEnd, latLng, data) {
+        console.log("addRouteMarker");
+
+        // == geocoding request format: /geocoding/v5/{mode}/{query}.json
+        var baseUrl = "https://api.mapbox.com/geocoding/v5/";
+        var routeMode = "mapbox.places/";
+        var latLngQuery = latLng[1] + "," + latLng[0];
+        var token = "pk.eyJ1IjoiYWx1bHNoIiwiYSI6ImY0NDBjYTQ1NjU4OGJmMDFiMWQ1Y2RmYjRlMGI1ZjIzIn0.pngboKEPsfuC4j54XDT3VA";
+        var url = baseUrl + routeMode + latLngQuery + ".json?access_token=" + token;
+
+        // == get directions for selected places
+        $.ajax({
+            url: url,
+            method: "GET",
+            dataType: "text"
+        }).done(function(jsonData){
+            console.log("*** ajax success ***");
+            var jsonData2 = $.parseJSON(jsonData);
+            console.dir(jsonData2);
+            updateRouteData(jsonData2);
+            makeRouteMarker(jsonData2);
+        }).fail(function(){
+            console.log("*** ajax fail ***");
```

Comment:
Need a user-friendly UI error here instead of a console.log. Can punt this to a later PR.